src/include/function/cast/cast_functions.h

```diff
@@ -87,27 +88,81 @@ inline std::string CastToString::castToStringWithVector(
 
 template<typename SRC, typename DST>
 static inline void numericDownCast(SRC& input, DST& result, const std::string& dstTypeStr) {
-    if (input < std::numeric_limits<DST>::min() || input > std::numeric_limits<DST>::max()) {
+    if ((signed)input < std::numeric_limits<DST>::min() ||
+        (signed)input > std::numeric_limits<DST>::max()) {
+        auto a = std::numeric_limits<DST>::min();
+        auto b = std::numeric_limits<DST>::max();
         throw common::RuntimeException(
             "Cast failed. " + std::to_string(input) + " is not in " + dstTypeStr + " range.");
     }
     result = (DST)input;
 }
 
+template<typename T>
+static inline bool tryCastStringToNum(const char* input, const uint64_t& len, T& result) {
+    // skip leading spaces
+    uint64_t start = 0;
+    while (start < len && isspace(input[start])) {
+        start++;
+    }
+
+    // skip trailing spaces
+    uint64_t end = len - 1;
+    while (end > start && isspace(input[end])) {
+        end--;
+    }
+    end++;
+
+    auto out = std::from_chars(input + start, input + end, result);
+    if (out.ec != std::errc{} || out.ptr != input + end) {
+        return false;
+    };
+    return true;
+}
+
+template<typename T>
+static inline T castStringToNum(
+    const char* input, const uint64_t& len, const std::string& type = "") {
```

Comment:
Instead of passing the dataType string, you can directly pass the LogicalType here. And use `LogicalTypeUtils::dataTypeToString(type)`